### PR TITLE
[3.3.3] Upgrade enrichment attributes rule node

### PR DESCRIPTION
**Pull request by van-vanich**
In the _Originator attributes_ rule node we can:
 - use ${metadataKey} for value from metadata, $[messageKey] for value from the message body.

But we can add this feature for this rule node:
 - customer attribute
 - tenant attribute
 - related attribute

So in this PR, I added this feature for this rule node. Also, refactoring tests for customer attributes and adding tests for tenant and related attribute rule nodes

### CODE REVIEW

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/metadata/TbGetRelatedAttributeNodeTest.java [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyCollection;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.thingsboard.rule.engine.api.TbRelationTypes.FAILURE;
+import static org.thingsboard.server.common.data.DataConstants.SERVER_SCOPE;
+
+@RunWith(MockitoJUnitRunner.Silent.class)
```
Comment (dmytro-landiak) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/metadata/TbGetRelatedAttributeNodeTest.java:71]: @RunWith(MockitoJUnitRunner.Silent.class)
Do we actually need this? If there are no good reasons to have it, then please remove Silent impl. The usage of this is not recommended.

File: rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/metadata/TbEntityGetAttrNode.java [hunk 2]
```diff
@@ -65,27 +67,35 @@ private void safeGetAttributes(TbContext ctx, TbMsg msg, T entityId) {
...
-        withCallback(config.isTelemetry() ? getLatestTelemetry(ctx, entityId) : getAttributesAsync(ctx, entityId),
+        withCallback(config.isTelemetry() ? getLatestTelemetry(ctx, entityId, msg) : getAttributesAsync(ctx, entityId, msg),
                 attributes -> putAttributesAndTell(ctx, msg, attributes),
                 t -> ctx.tellFailure(msg, t), ctx.getDbCallbackExecutor());
     }
 
-    private ListenableFuture<List<KvEntry>> getAttributesAsync(TbContext ctx, EntityId entityId) {
-        ListenableFuture<List<AttributeKvEntry>> latest = ctx.getAttributesService().find(ctx.getTenantId(), entityId, SERVER_SCOPE, config.getAttrMapping().keySet());
+    private ListenableFuture<List<KvEntry>> getAttributesAsync(TbContext ctx, EntityId entityId, TbMsg msg) {
+        List<String> attrProcess = TbNodeUtils.processPatterns(List.copyOf(config.getAttrMapping().keySet()), msg);
```
Comment (dmytro-landiak) [inline rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/metadata/TbEntityGetAttrNode.java:76]: Maybe move this out to the safeGetAttributes method. Create the map there and pass the needed to the getLatestTelemetry, getAttributesAsync, and putAttributesAndTell methods.

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/metadata/TbGetRelatedAttributeNodeTest.java [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+    @Mock
+    private RelationService relationService;
+    private TbMsg msg;
+    private Map<String, String> metaData;
+    private EntityRelation entityRelation;
+
+    @Before
+    public void init() throws TbNodeException {
+        TbGetRelatedAttrNodeConfiguration config = new TbGetRelatedAttrNodeConfiguration();
+        config = config.defaultConfiguration();
```
Comment (dmytro-landiak) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/metadata/TbGetRelatedAttributeNodeTest.java:98]: remove this line

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/metadata/TbGetRelatedAttributeNodeTest.java [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+        when(ctx.getDeviceService()).thenReturn(deviceService);
+        when(deviceService.findDeviceByIdAsync(any(), eq(deviceId))).thenReturn(Futures.immediateFuture(device));
+
+        entityAttributeFetched(customerId);
+    }
+
+    @Test
+    public void deviceCustomerTelemetryFetched() throws TbNodeException {
+        TbGetRelatedAttrNodeConfiguration config = new TbGetRelatedAttrNodeConfiguration();
+        config = config.defaultConfiguration();
```
Comment (dmytro-landiak) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/metadata/TbGetRelatedAttributeNodeTest.java:252]: remove this line

**Comment by ikulikov** [comment]
Do not forget to make pull request to rule nodes configuration UI and rule nodes docs.

**Pull request merged**